custom_components/matter_experimental/binary_sensor.py

```diff
@@ -56,18 +60,28 @@ def _update_from_device(self) -> None:
         self._attr_is_on = occupancy & 1 == 1
 
 
+@dataclass
+class MatterBinarySensorEntityDescription(
+    BinarySensorEntityDescription,
+    MatterEntityDescription,
+):
+    """Matter Binary Sensor entity description."""
+
+
+# You can't set default values on inherited data classes
+MatterSensorEntityDescriptionFactory = partial(
+    MatterBinarySensorEntityDescription, key=None, entity_cls=MatterBinarySensor
```

Comment:
Why do we not set a key? Shouldn't that be the device type?

Reply:
Oh, yeah that's possible.